### ISSUE TITLE
"realnic=default" option for pcap networking

### ISSUE THREAD

**Issue by pgalbraith** [joncampbell123/dosbox-x#3457]
### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

### Have you checked that no other similar feature request(s) already exists?

- [X] I have searched and didn't find any similar issues.

### Is your feature request related to a problem? Please describe.

Looking up and hand-assigning a pcap interface is a PITA :-)

### What you want

It would be really nice to allow "realnic=default" to be specified, and then let DOSBox-X try to figure out which interface has the default networking route (looking up the outgoing interface when attempting to reach a root DNS server will yield a decent choice of default interface in most cases, I would expect).

**Comment by Jookia** [comment]
From what I can see you can only listen to all interfaces at once on Linux, so you would need to effectively loop through each interface and send some kind of probe.

Maybe some code that opens all pcap interfaces with a filter on some packet, then send a packet somehow, then check all the filters for packets?

If we're crafting a packet, using a small TTL would mean they don't go in to the Internet, just to the local net.

**Comment by pgalbraith** [comment]
There's almost certainly a way to find the next-hop interface that would be used for reaching any given IP address.  The underlying TCP/IP stack obviously knows how to look this up, and almost certainly that will be exposed in some fashion to a programmer.  I know how to do this in java, and I doubt that the JDK is doing under the covers by explicitly trying every interface to see if it would route a given IP packet.

**Comment by Jookia** [comment]
I replied in email with a response but it seemed to get eaten. Yes, you could look up routing tables and it'd likely be less error prone. But you'd need to figure out how to map that to pcap devices.

**Comment by Jookia** [comment]
Yes, you can look this up by querying the routing tables of a system and
finding the default gateway.
